text/0000-near-shell.md

```diff
@@ -0,0 +1,313 @@
+
+Sub-commands for `network` include:
+
+* `near network list` : Display the current list of networks for the current instance of NEAR Shell.  
+* `near network status` :  Show the user's current network configuration.
+* `near network select` : Select a NEAR network as the default network for subsequent NEAR Shell commands as well as the network configuration for a local server validator node managed by this instance of NEAR Shell.
+* `near network add` : Add a network to the users's `~/.near` config files.
+* `near network remove` : Remove a network from tthe `~/.near` config files.
+* `near network monitor` : Interactively monitor one or more NEAR networks in a curses-like interface updated in a specified interval, default of 1s.
+* `near network help` : Display help on network subcommands. 
+
+### `near account <command>`
+
+This category is used to create, select, and configure accounts on NEAR networks for a given NEAR Shell user.  
+
+* `near account list`   : Display list of accounts configured on local host.
+* `near account status` : Display status of active or specified account, including token amounts, locked, etc.
+* `near account send`   : Send tokens from active or specified account to another NEAR account or contract.
+* `near account create` : Create a new account or sub-account.
+* `near account delete` : Delete an account or sub-account.
+* `near account create-key` : Delete an account or sub-account.
+* `near account revoke-key` : Delete an account or sub-account.
+* `near account select` : Select an account from the list of locally configured accounts as the active account.
+* `near account login`  : Log in the current active or specified account.
+* `near account logout` : Log out the current active or specified account. Essentially revoking a full access key.
+* `near account help`   : Display help on account subcommands.
+
+### `server`
+
+These commands are used to administer a NEAR server validator node.  Currently, only one server per NEAR Shell is proposed; however, multiple servers per physical host should be considered.
+
+* `near server status`  : Display status of NEAR validator server on current host.
+* `near server start`   : Start NEAR validator server on current host.
+* `near server stop`    : Stop NEAR validator server on current host.
+* `near server monitor` : Interactively display NEAR validator server status on current host.
+* `near server tail`    : Tail the log of the NEAR validator server on current host.
+* `near server help`    : Display help on server subcommands.
+* `near server stake`   : Stake tokens on the configured network with the active or specified account
+
+### `near contract <command>` 
+* `near contract list`
+* `near contract status`
+* `near contract add`
+* `near contract remove`
+* `near contract build`
```

Comment:
Just highlighting a discussion here: https://github.com/nearprotocol/near-shell/pull/275

Since the build process for AssemblyScript and Rust differs, it may make sense to either:

1) Remove this and lean on the tools for building in the language ecosystems
2) In implementation, the build process has a step where the language is determined.

@vgrichina makes good points for (1) here: https://github.com/nearprotocol/near-shell/pull/275#issuecomment-595419971

Reply:
There is another important aspect to this. 
We don't currently have support for multi-contract projects.

And I don't think we can easily provide anything like `near build` for these, as they can easily be mix of Rust / AssemblyScript / whatever else contracts which all need to be deployed together.

We need to think of some build system supporting dependencies (i.e. something like `make` or `gulp`) to put this all together and provide ability for developer to customize individual steps as necessary.

Reply:
> We need to think of some build system supporting dependencies (i.e. something like make or gulp) to put this all together
`near build` currently runs gulp. if the suggestion is to remove `near build` would we be adding this back in?
https://github.com/nearprotocol/near-shell/blob/7d9e8601b0eeb05872406e8bf5eca3e14098224c/bin/near-cli.js#L118-L122

Reply:
I think we should:
1) remove `near build`
2) have design for multi contract projects if we want to move away from gulp

Reply:
Removed `near build` from the spec
Haven't added a recommended method for multi-contract projects, as I think we haven't discussed the alternatives yet. Honestly, I don't have a good gut instinct for that task and would like to hear from others.

Reply:
How about something like [lerna](https://lerna.js.org/)? We could probably abstract away all of the lerna operations with `near` commands.

Reply:
@Sladuca Lerna solves somewhat orthogonal problem, I'm talking more about stuff how you can have one project with multiple deployments (e.g. given this is shared system you may want to deploy single contract on multiple accounts). 

Other than that I think it should be possible to use Lerna with near-shell as an option, no need to wrap around with near-shell commands. We want to provide dev tools that fit into different workflows (i.e. not tied to specific build tools ideally) + provide reasonable default workflows in examples.